src/main/java/Ping/Parser.java

```diff
@@ -0,0 +1,196 @@
+        }
+    }
+
+    private static Command parseUnmark(String[] restCommands) throws PingException {
+        try {
+            int i = Integer.parseInt(restCommands[1]) - 1;
+            return new UnmarkCommand(i);
+        } catch (Exception e) {
+            throw new PingException("Invalid number");
+        }
+    }
+
+    private static Command parseTodo(String[] todoCommand) throws PingException {
+        StringBuilder rest = new StringBuilder();
+        try {
+            for (int i = 1; i < todoCommand.length; i++) {
+                rest.append(todoCommand[i]).append(" ");
+            }
+            if (rest.length() > 0) {
+                return new AddCommand(new Todo(rest.toString()));
+            } else {
+                throw new PingException("Did you type right?");
+            }
+        } catch (Exception e) {
+            throw new PingException("Incorrect number or command");
+        }
+    }
+
+    private static Command parseDelete(String[] delCommand) throws PingException {
+        try {
+            int i = Integer.parseInt(delCommand[1]) - 1;
+            return new DeleteCommand(i);
+        } catch (Exception e) {
+            throw new PingException("Invalid number");
+        }
+        
+    }
+
+    private static Command parseDeadline(String[] dlCommand) throws PingException {
+        StringBuilder rest = new StringBuilder();
+        StringBuilder date = new StringBuilder();
+        try {
+            int idx = 0;
+            for (int i = 1; i < dlCommand.length; i++) {
+                if (dlCommand[i].equals("/by")) {
+                    idx = i;
+                    break;
+                } else {
+                    rest.append(dlCommand[i]).append(" ");
+                }
+            }
+            // Check for weekdays or month
+            int check = idx + 1;
+            for (int j = idx + 1; j < dlCommand.length; j++) {
+                if (check != dlCommand.length - 1) {
+                    date.append(dlCommand[j]).append(" ");
+                    check++;
+                } else {
+                    date.append(dlCommand[j]);
+                }
+            }
+
+            LocalDate dt = DateTimeCheck.timeCheckOnDate(date.toString());
+            Deadline dl = new Deadline(rest.toString(), dt);
+            if (rest.length() > 0 && dt != null) {
+                return new AddCommand(dl);
+            } else {
+                throw new PingException("Did you type right?");
+            }
+        } catch (PingException e) {
+            throw new PingException(e.getMessage());
+        } catch (Exception e) {
+            throw new PingException("Incorrect number or command");
+        }
+    }
+
+    private static Command parseEvent(String[] evCommand) throws PingException {
+        StringBuilder rest = new StringBuilder();
+        StringBuilder date1 = new StringBuilder();
+        StringBuilder date2 = new StringBuilder();
```

Comment:
Perhaps these variables can be renamed to be more meaningful according to the naming standards.
```suggestion
        StringBuilder startDate = new StringBuilder();
        StringBuilder endDate = new StringBuilder();
```